core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ShaclSailConnection.java

```diff
@@ -166,10 +169,12 @@ public void commit() throws SailException {
 			sail.setNodeShapes(nodeShapes);
 		}
 
-		if (sail.holdsWriteLock(writeLockStamp)) {
```

Comment:
writeLockStamp should never both be "not null" and "not active". However we still double check, since locking is something that is easy to get wrong in subtle and unpredictable ways.